heft-plugins/heft-webpack4-plugin/src/shared.ts

```diff
@@ -23,25 +23,40 @@ export type IWebpackConfiguration =
 /**
  * @public
  */
-export interface IWebpackBundleSubstageProperties extends IBundleSubstageProperties {
+export interface IWebpackVersions {
+  webpackVersion: string;
+  webpackDevServerVersion: string;
+}
+
+/**
+ * @public
+ */
+export interface IWebpackPluginAccessor {
   /**
-   * The configuration used by the Webpack plugin. This must be populated
-   * for Webpack to run. If webpackConfigFilePath is specified,
-   * this will be populated automatically with the exports of the
-   * config file referenced in that property.
+   * A hook that provides the versions of Webpack and Webpack Dev Server.
+   */
+  readonly onEmitWebpackVersionsHook?: AsyncParallelHook<IWebpackVersions>;
+  /**
+   * A hook that allows for modification of the  configuration used by the Webpack
+   * plugin. This must be populated for Webpack to run. If webpackConfigFilePath is
+   * specified, this will be populated automatically with the exports of the config
+   * file referenced in that property.
    *
    * @remarks
    * Tapable event handlers can return `null` instead of `undefined` to suppress
    * other handlers from creating a configuration object.
    */
   // We are inheriting this problem from Tapable's API
   // eslint-disable-next-line @rushstack/no-new-null
```

Comment:
We could just as easily use `false` or `0` or other value as `null`. The `tapable` waterfall API simply allows for handlers to return `undefined` if they aren't modifying the value.